compiler/rustc_interface/src/passes.rs

```diff
@@ -883,6 +909,10 @@ fn run_required_analyses(tcx: TyCtxt<'_>) {
                 CStore::from_tcx(tcx).report_unused_deps(tcx);
             },
             {
+                if tcx.crate_types().contains(&CrateType::Sdylib) {
```

Comment:
This works accidentally because all the tests use `#![crate_type = "sdylib"]`.
I think it's supposed to use `if is_inteface_buid` instead.

The `is_inteface_buid` optimization can be put inside the `exportable_items` and `stable_order_of_exportable_impls` queries, then we won't need to do it explicitly neither here nor in metadata encoder.